docker/utils/utils.py

```diff
@@ -100,7 +101,14 @@ def tar(path, exclude=None, dockerfile=None, fileobj=None):
     exclude = exclude or []
 
     for path in sorted(exclude_paths(root, exclude, dockerfile=dockerfile)):
-        t.add(os.path.join(root, path), arcname=path, recursive=False)
+        i = t.gettarinfo(os.path.join(root, path), arcname=path)
+        if sys.platform == 'win32':
+            i.mode = i.mode & 0o755 | 0o111
+        if i.isreg():
+            with open(os.path.join(root, path), 'rb') as f:
+                t.addfile(i, f)
```

Comment:
I don't think this `if i.isref()` branch is necessary. I think the warning in the docs is only for the case where you're passing a `fileobj`. I don't think it means that it is required to pass it on windows.

Did you run into problems without it?